std/datetime.d

```diff
@@ -30404,70 +30496,140 @@ unittest
+
+    On most systems, unix time is equivalent to time_t, but that depends on the
+    C implementation as the exact representation of time_t is
+    implementation-defined (e.g. it does not appear that the Digital Mars C
+    runtime uses unix time for time_t; it appears to apply the local DST to
+    time_t).
+
+    "std time"'s epoch is based on the Proleptic Gregorian Calendar per ISO
+    8601 and is what $(LREF SysTime) uses internally. However, holding the time
+    as an integer in hnescs since that epoch technically isn't actually part of
+    the standard, much as it's based on it, so the name "std time" isn't
+    particularly good, but there isn't an official name for it. C# uses "ticks"
+    for the same thing, but they aren't actually clock ticks, and the term
+    "ticks" $(I is) used for actual clock ticks for $(CXREF time, MonoTime), so
+    it didn't make sense to use the term ticks here. So, for better or worse,
+    std.datetime uses the term "std time" for this.
+
+    By default, the return type is time_t (which is normally an alias for
+    int on 32-bit systems and long on 64-bit systems), but if a different
+    size is required than either int or long can be passed as a template
+    argument to get the desired size.
+
+    If the return type is int, and the result can't fit in an int, then the
+    closest value that can be held in 32 bits will be used (so $(D int.max)
+    if it goes over and $(D int.min) if it goes under). However, no attempt
+    is made to deal with integer overflow if the return type is long.
 
     Params:
+        T = The return type (int or long). It defaults to time_t, which is
+            normall 32 bits on a 32-bit system and 64 bits on a 64-bit
+            system.
         stdTime = The std time to convert.
+
+    Returns:
+        A signed integer representing the unix time which is equivalent to
+        the given std time.
+
+    See_Also:
+        SysTime.toUnixTime
   +/
-time_t stdTimeToUnixTime(long stdTime) @safe pure nothrow
+T stdTimeToUnixTime(T = time_t)(long stdTime) @safe pure nothrow
+    if(is(T == int) || is(T == long))
 {
     immutable unixTime = convert!("hnsecs", "seconds")(stdTime - 621_355_968_000_000_000L);
 
-    static if(time_t.sizeof >= long.sizeof)
-        return cast(time_t)unixTime;
-    else
+    static assert(time_t.sizeof == 4 || time_t.sizeof == 8,
+                  "A system where time_t isn't 32-bit or 64-bit? That's not currently supported.");
+
+    static if(is(T == long))
+        return unixTime;
+    else static if(is(T == int))
     {
-        if(unixTime > 0)
-        {
-            if(unixTime > time_t.max)
-                return time_t.max;
-            return cast(time_t)unixTime;
-        }
+        if(unixTime > int.max)
+            return int.max;
+        return unixTime < int.min ? int.min : cast(int)unixTime;
+    }
+    else static assert(0, "How is it that T isn't int or long?");
```

Comment:
This message is not very helpful (if seeing as the result of some build). Either change it to something more definitive (e.g. `"stdTimeToUnixTime does not support return type " ~ T.stringof`) or remove.